docs/running-on-kubernetes.md

```diff
@@ -990,6 +990,52 @@ See the [configuration page](configuration.html) for information on Spark config
   Specify whether executor pods should be deleted in case of failure or normal termination.
   </td>
 </tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.submission.connectionTimeout</code></td>
+  <td>10000</td>
+  <td>
+    Connection timeout in milliseconds for the kubernetes client to use for starting the driver. In client mode, use
+    <code>spark.kubernetes.kubernetesClient.connectionTimeout</code> instead.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.submission.requestTimeout</code></td>
+  <td>10000</td>
+  <td>
+    Request timeout in milliseconds for the kubernetes client to use for starting the driver. In client mode, use
+    <code>spark.kubernetes.kubernetesClient.submission.requestTimeout</code> instead.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.driver.connectionTimeout</code></td>
+  <td>10000</td>
+  <td>
+    Connection timeout in milliseconds for the kubernetes client in driver to use when requesting executors.
+    In client mode, use <code>spark.kubernetes.kubernetesClient.connectionTimeout</code> instead.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.driver.requestTimeout</code></td>
+  <td>10000</td>
+  <td>
+    Request timeout in milliseconds for the kubernetes client in driver to use when requesting executors.
+    In client mode, use <code>spark.kubernetes.kubernetesClient.connectionTimeout</code> instead.
+  </td>
+</tr>
+<tr>
+  <td><code>spark.kubernetes.kubernetesClient.connectionTimeout</code></td>
```

Comment:
Why you need separate options for client mode? IMO, they are only necessary when they can be used in conjunction with the two new options for cluster mode, which won't be the case.

Reply:
+1